File: 2486-Append Characters to String to Make Subsequence.py
class Solution(object):
    def appendCharacters(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: int
        """
        left = 0
        
        for i in range(len(s)):
            if left < len(t) and s[i] == t[left]:
                left += 1
                
        if left != len(t):
            return len(t) - left
        
        else:
            return 0

File: C++/2486-Append Characters to String to Make Subsequence.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        int i=0, j=0;
        for (int i=0, N=s.size(); i<N; i++) {
            if (s[i] == t[j]) j++;
        }
        return t.size() - j;
    }
};